compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParser.java

```diff
@@ -106,6 +106,10 @@ protected Solution recover(STToken token, ParserRuleContext currentCtx, Object..
         Solution sol = this.errorHandler.recover(currentCtx, token, args);
         // If the action is to remove, then re-parse the same rule.
         if (sol.action == Action.REMOVE) {
+            if (sol.isFailSafe) {
+                // Clean up the inserted token, which cannot be parsed.
+                this.insertedToken = null;
+            }
```

Comment:
We don't need this check right?. If the solution is `REMOVE` then we need to clean `this.insertedToken` anyway

Reply:
+1